runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -186,6 +186,12 @@ NNFW_STATUS nnfw_session::run()
 NNFW_STATUS nnfw_session::set_input(uint32_t index, NNFW_TYPE /*type*/, const void *buffer,
                                     size_t length)
 {
+  if (buffer == nullptr)
+  {
+    std::cerr << "Error during nnfw_session::set_input : given buffer is NULL" << std::endl;
+    return NNFW_STATUS_ERROR;
+  }
```

Comment:
Does nnfw api not allow calling this api with index of optional input?

Reply:
>But I had second thoughts - let the users to set nullptr buffer with size 0, and it is used to specify an optional input.

Good point. User can reset buffer setting by this. (ex. use general input on first inference, then reset by `nullptr` for optional input on second inference) Different with `NNAPI`, we don't have independent object to represent each inference on same session.

Reply:
I see. It looks good.